spec/models/miq_widget_spec.rb

```diff
@@ -130,18 +130,18 @@
       end
 
       it "ignores the legacy format admin|db_name" do
-        expect { FactoryBot.create(:miq_widget_set, :set_data_with_one_widget, :set_data_with_one_widget, :name => "#{@user1.userid}|Home", :owner => @user1.current_group) }.to raise_error(ActiveRecord::RecordInvalid)
+        expect { FactoryBot.create(:miq_widget_set, :name => "#{@user1.userid}|Home", :owner => @user1.current_group) }.to raise_error(ActiveRecord::RecordInvalid)
       end
 
       context 'with subscribers' do
         before do
-          ws = FactoryBot.create(:miq_widget_set, :set_data_with_one_widget, :name => "Home", :userid => @user1.userid, :group_id => @group1.id)
+          ws = FactoryBot.create(:miq_widget_set, :name => "Home", :userid => @user1.userid, :group => @group1)
```

Comment:
ws = FactoryBot.create(:miq_widget_set, :name => "Home", :userid => @user1.userid, :owner => @user1, :group => @group1)

We need the owner set here too, that should fix three of the four failures.

Reply:
@lpichler it feels like passing in the group or owner in these tests should match what ever we expect the ui to do. Whether we require both variables or just one, let’s copy what we want users to do in the core code.